Assets/MixedRealityToolkit/Services/BaseExtensionService.cs

```diff
@@ -13,22 +13,35 @@ namespace Microsoft.MixedReality.Toolkit.Core.Services
     /// <remarks>
     /// Empty, but reserved for future use, in case additional <see cref="Interfaces.IMixedRealityExtensionService"/> properties or methods are assigned.
     /// </remarks>
-    public abstract class BaseExtensionService : BaseServiceWithConstructor, Interfaces.IMixedRealityExtensionService
+    public abstract class BaseExtensionService : BaseService, Interfaces.IMixedRealityExtensionService
     {
-        /// <summary>
-        /// Configuration Profile
-        /// </summary>
-        protected BaseMixedRealityProfile ConfigurationProfile { get; set; } = null;
-
         /// <summary>
         /// Constructor.
         /// </summary>
-        /// <param name="name">Friendly name of the service.</param>
-        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
-        /// <param name="profile">The service's configuration profile.</param>
-        public BaseExtensionService(string name, uint priority, BaseMixedRealityProfile profile) : base(name, priority)
+        /// <param name="registrar">The <see cref="IMixedRealityServiceRegistrar"/> instance that loaded the service.</param>
+        /// <param name="name">The friendly name of the service.</param>
+        /// <param name="priority">The registration priority of the service.</param>
+        /// <param name="profile">The configuration profile for the service.</param>
+        public BaseExtensionService(
+            IMixedRealityServiceRegistrar registrar, 
+            string name = null, 
+            uint priority = DefaultPriority, 
+            BaseMixedRealityProfile profile = null) : base()
         {
+            Registrar = registrar;
+            Name = name;
```

Comment:
Similar to the system cache of this, there is no guarantee this won't be null, but it is accessed in many locations under the assumption its never null.